cfy_cluster_setup/main.py

```diff
@@ -291,13 +298,57 @@ def generate_config(output_path):
                 output_path)
 
 
+def _check_path(config, key):
+    path = config.get(key)
+    if (not path) or (not exists(expanduser(path))):
+        raise ValidationError('Path {0} does not exist'.format(path))
+
+    config[key] = expanduser(path)
+
+
+def _check_value_supplied(config, key):
+    if not config.get(key):
+        raise ValidationError('{0} is not specified'.format(key))
+
+
+def _check_existing_vms(config):
+    ca_cert_path = config.get('ca_cert_path')
+    if ca_cert_path:
+        _check_path(config, 'ca_cert_path')
+
+    existing_vms_list = config.get('existing_vms')
+    for vm_name, vm_dict in existing_vms_list.items():
+        logger.info('Validating %s', vm_name)
+        if (not vm_dict['private_ip']) or (not vm_dict['public_ip']):
+            raise ValidationError('private_ip and public_ip should be '
+                                  'specified for all instances')
+        if ca_cert_path:
+            cert_path = vm_dict.get('cert_path')
+            key_path = vm_dict.get('key_path')
+            if (not cert_path) or (not key_path):
+                raise ValidationError('cert_path and key_path should be '
+                                      'specified for all instances')
+            check_cert_key_match(cert_path, key_path)
+            check_signed_by(ca_cert_path, cert_path)
+            check_san(vm_name, vm_dict, cert_path)
+
+
+def _validate_config(config):
+    _check_path(config, 'key_file_path')
```

Comment:
If you did want to collect all errors then raise once, making these check functions return any errors found and appending them to a list then raising a '\n'.joined list at the end (if errors:) might be reasonable?

Reply:
Correct, I'll apply this mechanism.